### PR TITLE
[binding] add new api and refine codes

### PR DESCRIPTION
**Pull request by cdliang11**
1. Add new api:

    - compute_cosine_score
    - extract_embedding_feat

2. Rename `Inference` to `Speaker`.

### CODE REVIEW

File: runtime/binding/python/py/inference.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 import kaldiio
```
Comment (robin1001) [inline runtime/binding/python/py/inference.py:22]: rename the file to speaker.py

File: runtime/binding/python/py/inference.py [hunk 2]
```diff
@@ -80,13 +80,29 @@ def extract_embedding_wav(self, wav_path: str, resample_rate: int = 16000):
...
+            embeddings(list): [B, emb_dim]
+        """
+        assert isinstance(
+            feats, np.ndarray), 'NOTE: the type of feats need be np.ndarray.'
+        assert len(feats.shape) == 3, "NOTE: the shape of feats is [B, T, D]."
+        embeddings = self.session.run(output_names=['embs'],
+                                      input_feed={'feats': feats})
+        return embeddings[0]  # [B, embed]
+
     def extract_embedding(self,
```
Comment (robin1001) [inline runtime/binding/python/py/inference.py:106]: This is the most important interface. I think we should keep it for extracting embedding and return numpy ndarrary, which is preferred for most users.

And we can add something like `extract_embedding_kaldi`, in which the input and output are in kaldi style(scp/ark)

File: runtime/binding/python/README.md [hunk 3]
```diff
@@ -47,15 +47,42 @@ The parameters of `extract_embedding_wav`
...
+speaker.extract_embedding(wav_scp, 'embed.ark')
 ```
 
-The parameters of `extract_embedding`
+The parameters of `extract_embedding`:
 - `wav_path` (str): the path of wav
 - `embed_ark` (str): the path of `$ouput`.ark
 - `resample_rate` (int): resampling rate. Default: 16000
 
+### Extract embedding from feature(fbank)
```
Comment (robin1001) [inline runtime/binding/python/README.md:59]: The readme should be refactored if `extract_embedding` is kept as ndarray interface.

Here is the recommended order the outline.
1.  Extract speaker embedding
2.  Compute speaker simliarity
3.  Optional, Extract speaker embedding with fbank feature
4. Optional, Extract speaker embedding with kaldi IO

**Review by robin1001 — CHANGES_REQUESTED** [review]
please see inline.

**Comment by cdliang11** [comment]
> please see inline.

done.

**Review by robin1001 — APPROVED** [review]

**Comment by robin1001** [comment]
Great job!

**Comment by robin1001** [comment]
Great job!

**Comment by cdliang11** [comment]
Uploaded to [pypi](https://pypi.org/project/wespeakerruntime/), and the [huggingface demo](https://huggingface.co/spaces/wenet/wespeaker_demo) has been updated.

**Pull request merged**